emmet-core/emmet/core/openmm/calculations.py

```diff
@@ -0,0 +1,71 @@
+from typing import List, Optional, Union
+from datetime import datetime
+
+from pydantic import BaseModel, Field
+
+from emmet.core.openmm.tasks import Calculation
+
+
+class CalculationsDoc(BaseModel):
+    task_names: List[str] = Field(None, description="Names of tasks.")
+
+    calc_types: List[str] = Field(None, description="Types of calculations.")
+
+    elapsed_times: List[Union[float, None]] = Field(
+        None, description="Elapsed time for each calculation."
+    )
+
+    steps: List[Union[float, None]] = Field(
+        None, description="n_steps for each calculation."
+    )
+
+    step_sizes: List[Union[float, None]] = Field(
+        None, description="Step sizes for each calculation."
+    )
+
+    temperatures: List[Union[float, None]] = Field(
+        None, description="Temperature for each calculation."
+    )
+
+    pressures: List[Union[float, None]] = Field(
+        None, description="Pressure for each calculation."
+    )
+
+    friction_coefficients: List[Union[float, None]] = Field(
+        None, description="Friction coefficients for each calculation."
+    )
+
+    completed_at: Optional[datetime] = Field(
+        None,
+        description="Timestamp for when the final calculation completed.",
+    )
+
+    job_uuid: Optional[str] = Field(
+        None, description="The UUID of the flow that generated this data."
+    )
+
+    flow_uuid: Optional[str] = Field(
+        None, description="The UUID of the top level host from that job."
+    )
+
+    @classmethod
+    def from_calcs_reversed(
+        cls,
+        calcs_reversed: List[Calculation],
+        job_uuid: Optional[str] = None,
+        flow_uuid: Optional[str] = None,
+    ) -> "CalculationsDoc":
+        calcs = calcs_reversed[::-1]
```

Comment:
Is there a reason to un-reverse the calcs_reversed here? 

From this I understand that the entries in each of the arrays for the document fields will correspond to the order in which the calcs were completed, but would that be confusing given the calcs_reversed are in the opposite order? 

I see this is still taking the last calc (after un-reversing) for the completed_at field.

Reply:
I want the lists in `CalculationsDoc` to follow a sequential order rather than reversed because it's more intuitive. The "reversal" of `calcs_reversed` is a bit less intuitive IMO (as evidenced by it being called `calcs_reversed` rather than `calcs`).

I don't want to have `task_names_reversed`, `calc_types_reversed`, etc.

Reply:
Yep, agreed,  makes sense. But a small doc string might be helpful since all the other workflows, doc models, and builders do follow the calcs_reversed pattern

Reply:
Added a docstring to clarify.